lib/cassandra/protocol/coder.rb

```diff
@@ -370,6 +370,7 @@ def write_value_v3(buffer, value, type)
         when :double           then write_double(buffer, value)
         when :float            then write_float(buffer, value)
         when :int              then write_int(buffer, value)
+        when :smallint         then write_smallint(buffer, value)
```

Comment:
This is likely what I've fixed in https://github.com/datastax/ruby-driver/commit/a168e8c55f6bef60100bfc1bb014eaeb72654f7e